dwave/system/composites/reversecomposite.py

```diff
@@ -0,0 +1,204 @@
+                b = sampleset.record[sampleset.record.energy == gse]
+                b.sort(order='num_occurrences')
+                initial_state = dict(zip(sampleset.variables, b[-1].sample))
+            else:
+                # if not reinitialized, take the last state as the next initial state
+                initial_state = dict(zip(sampleset.variables, sampleset.record.sample[-1]))
+
+        samples = vectors.pop('sample')
+        return dimod.SampleSet.from_samples((samples, bqm.variables),
+                                bqm.vartype,
+                                info={'anneal_schedules': anneal_schedules},
+                                **vectors)
+
+
+class BatchReverseComposite(dimod.ComposedSampler):
+    """ Composite that accepts multiple samples to initialize from.
+
+    Args:
+       sampler (:obj:`dimod.Sampler`):
+            A dimod sampler
+
+    """
+
+    def __init__(self, child_sampler):
+        self._children = [child_sampler]
+
+    @property
+    def children(self):
+        return self._children
+
+    @property
+    def parameters(self):
+        param = self.child.parameters.copy()
+        param['initial_states'] = []
+        return param
+
+    @property
+    def properties(self):
+        return {'child_properties': self.child.properties.copy()}
+
+    def sample(self, bqm, **parameters):
+        """ Composite that accepts multiple initial states to reverse annealing from
+
+        Args:
+            bqm (:obj:`dimod.BinaryQuadraticModel`):
+                Binary quadratic model to be sampled from.
+
+            **parameters:
+                Parameters for the sampling method, specified by the child sampler.
+
+        Returns:
+            :obj:`dimod.SampleSet` that has initial_state and schedule_index fields in addition to
+            'energy' and 'num_occurrences'
+
+        """
+        child = self.child
+
+        if 'initial_states' not in parameters:
+            return child.sample(bqm, **parameters)
+
+        initial_states = parameters.pop('initial_states')
+
+        # there is gonna be way too much data generated - better to histogram them
+        parameters['answer_mode'] = 'histogram'
+
+        # prepare data fields for the new sampleset object
+
+        vectors = {}
+        for initial_state in initial_states:
+
+            if not isinstance(initial_state, dict):
+                initial_state = dict(zip(bqm.variables, initial_state))
+
+            sampleset = child.sample(bqm, initial_state=initial_state, **parameters)
+            initial_state_, _ = dimod.as_samples(initial_state)
+            vectors = _update_data_vector(vectors, sampleset,
+                                          {'initial_state': [initial_state_[0]] * len(sampleset.record.energy)})
+
+        samples = vectors.pop('sample')
+
+        return dimod.SampleSet.from_samples((samples, bqm.variables),
+                                bqm.vartype,
+                                info={},
```

Comment:
In the other composite you did, ```info={'anneal_schedules': anneal_schedules},```, did you want to return the initial states in the info field here?

Reply:
in this composite I'm returning schedule index instead of the schedule itself and thats why I return it in the info field for this. It can also be an optional return. The other one returns the initial_state itself.